python/kfserving/kfserving/storage.py

```diff
@@ -193,9 +193,10 @@ def _download_local(uri, out_dir=None):
     @staticmethod
     def _create_minio_client():
         # Remove possible http scheme for Minio
-        url = re.compile(r"https?://")
-        minioClient = Minio(url.sub("", os.getenv("S3_ENDPOINT", "")),
+        url = urlparse(os.getenv("S3_ENDPOINT", ""))
+        use_ssl = url.scheme=='https' if url.scheme else bool(os.getenv("USE_SSL", True))
```

Comment:
We have env variable `S3_USE_HTTPS`(https://github.com/kubeflow/kfserving/blob/master/pkg/controller/kfservice/resources/credentials/s3/s3_secret.go#L32)

Reply:
fixed!